grails-app/controllers/org/pih/warehouse/inventory/InventoryController.groovy

```diff
@@ -575,49 +575,70 @@ class InventoryController {
 
 
     def listExpiredStock() {
-        def location = Location.get(session.warehouse.id)
-        def categorySelected = (params.category) ? Category.get(params.category) : null
-        def inventoryItems = dashboardService.getExpiredStock(categorySelected, location)
-        def categories = inventoryItems?.collect { it.product.category }?.unique()
-        def quantityMap = productAvailabilityService.getQuantityOnHandByInventoryItem(location)
-        def expiredStockMap = [:]
-        inventoryItems.each { inventoryItem ->
-            expiredStockMap[inventoryItem] = quantityMap[inventoryItem]
+        Location location = Location.get(session.warehouse.id)
+        Category categorySelected = Category.get(params.category)
```

Comment:
Shouldn't we use data services since we are on Grails 3?

Reply:
Let's come up with our conventions for data services before we go lock, stock, and barrel. But what exactly would be changing here? 

Also, what is going to happen if params.category is null here?

And just for giggles, we *could* use Location, Category, Boolean as arguments to this action and be done with the data binding.

Reply:
@jmiranda in general the difference is that data service's read operations are run in a `readOnly` transaction. 
I once mentioned to the guys, that I needed to do a research about what happens if we are in a service (marked as `@Transactional`) and we'd run a method from data service that is `readOnly` - if the outer (service's) transaction would "swallow" that, hence the fetch would be done using our outer transaction, or if it would set our outer transaction as `readOnly` for a while.
In the end I think this `readOnly` doesn't matter at this point, because as soon as we run the:
```groovy
categoryDataService.get(params.category)
```
the readOnly transaction would commit at this point (so in further lines we could still update a category).

> Also, what is going to happen if params.category is null here?

The data service's get evaluates to the basic `Domain.get` anyway, hence it'd return `null` if an entity didn't exist.

I think the data services methods matter much, if we do not take care of a transaction lifecycle on our own, but in the end, they evaluate to basic domain methods like `Domain.get(...)`, `Domain.delete(...)`, `Domain.save(...)`, so I don't have a strong opinion about forcing to use data services for such simple cases.
I find the data services very helpful and powerful for stuff like fetch join query, etc. where it has helped us a lot already.